src/lib/holocene/maximizable/maximizable.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+  import { onMount, type Snippet } from 'svelte';
+  import { twMerge as merge } from 'tailwind-merge';
+
+  import MaximizeButton from './button.svelte';
+
+  interface Props {
+    children: Snippet;
+    maximized: boolean;
+    onMaximize: (maximized: boolean) => void;
+    class?: string;
+    enabled?: boolean;
+    actions?: Snippet;
+  }
+
+  let {
+    children,
+    maximized,
+    onMaximize,
+    class: className = undefined,
+    enabled = true,
+    actions = undefined,
+  }: Props = $props();
+
+  let escapeListener = $state((event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      onMaximize(false);
+    }
+  });
+
+  $effect(() => {
+    if (maximized) {
+      document.addEventListener('keydown', escapeListener);
+    } else {
+      document.removeEventListener('keydown', escapeListener);
+    }
+  });
+
+  onMount(() => {
+    // on unmount
+    return () => {
+      document.removeEventListener('keydown', escapeListener);
+    };
+  });
```

Comment:
Instead of managing the listener directly I would recommend using `<svelte:body />` it's dope and it will do it for you

https://svelte.dev/docs/svelte/svelte-body